commands/diagnostics.vegman

```diff
@@ -119,3 +119,50 @@ function cmd_traceroute {
 function cmd_netstat {
   netstat "$@"
 }
+
+# D-Bus intrusion objects
+INTR_SERVICE="xyz.openbmc_project.intrusionsensor.service"
+INTR_DBUS_SERVICE="xyz.openbmc_project.IntrusionSensor"
+INTR_DBUS_PATH="/xyz/openbmc_project/Intrusion/Chassis_Intrusion"
+INTR_DBUS_INTERFACE="xyz.openbmc_project.Chassis.Intrusion"
+INTR_DBUS_PROPERTY="Status"
+
+# @sudo cmd_intrusion_on admin,tech
+# @restrict cmd_intrusion_on admin,tech
+# @doc cmd_intrusion_on
+# Turn the chassis intrusion detection on
+function cmd_intrusion_on {
+  sysctemctl start ${INTR_SERVICE}
+}
+
+# @sudo cmd_intrusion_off admin,tech
+# @restrict cmd_intrusion_off admin,tech
+# @doc cmd_intrusion_off
+# Turn the chassis intrusion detection off
+function cmd_intrusion_off {
+  sysctemctl stop ${INTR_SERVICE}
+}
+
+# @sudo cmd_intrusion_show admin,tech
+# @restrict cmd_intrusion_show admin,tech
+# @doc cmd_intrusion_show
+# Show the chassis intrusion detection status
+function cmd_intrusion_show {
+  state=$(busctl get-property ${INTR_DBUS_SERVICE} "${INTRUSION_DBUS_PATH}" \
+          ${INTR_DBUS_INTERFACE} ${INTRUSION_DBUS_PROPERTY} \
+          | awk '{print $NF;}')
+  if [ "$state" == "Normal" ]; then
+    echo "Chassis is closed"
+  else
+    echo "Hardware intrusion"
+  fi
+}
+
+# @sudo cmd_intrusion_clear admin,tech
+# @restrict cmd_intrusion_clear admin,tech
+# @doc cmd_intrusion_clear
+# Clear the chassis intrusion status
+function cmd_intrusion_clear {
+  # GPIO pin A1 = 1
+  gpioset gpiochip0 1=1 && log_audit "Asserted chassis intrusion pin"
```

Comment:
Any idea how to distinguish between boards, only with proper GPIO pin naming?

Reply:
> isn't RST pin should be de-asserted to initial state?
no, as I see from doc.

Reply:
> Any idea how to distinguish between boards, only with proper GPIO pin naming?

No, have no idea... But you can use /etc/os-release

> no, as I see from doc.

What doc? What I see in task description, it says "assert for 100-500ms". I believe, it should be cleared, otherwise, how  you gonna to asset it next time?

Reply:
> No, have no idea... But you can use /etc/os-release

But, well... Maybe just use gpiofind to locate RST pin ans if find nothing - then say that it is not supported.

Reply:
Yea, gpiofind may distinguish it, rely on /etc/os-release seems not so good...